common/element_handle.go

```diff
@@ -253,15 +253,19 @@ func (h *ElementHandle) checkHitTargetAt(apiCtx context.Context, point Position)
 		return false, err
 	}
 
+	// Either we're done or an error happened (returned as "error:..." from JS)
+	const done = "done"
+	//nolint:forcetypeassert
 	value := result.(goja.Value)
-	switch value.ExportType().Kind() {
-	case reflect.String: // Either we're done or an error happened (returned as "error:..." from JS)
-		if value.String() == "done" {
-			return true, nil
-		}
+	if value.ExportType().Kind() != reflect.String {
+		// We got a { hitTargetDescription: ... } result
+		// Meaning: Another element is preventing pointer events.
+		return false, errorFromDOMError("error:intercept")
```

Comment:
Would it be safer to explicitly check for this result instead of assuming it's always an intercept error?

Reply:
But it's always an intercept error :)

Reply:
Is that based on your tests or on some documentation? In general it's better to be explicit, but I don't mind it, just checking.

Reply:
It's based on the JS code we use ;)

https://github.com/grafana/xk6-browser/blob/3ad187ce3f3cd07fd572bb504f8a69d28cf29d4f/common/js/injected_script.js#L279-L312

Reply:
Better?

```suggestion
	if value.ExportType().Kind() != reflect.String {
		// We got a { hitTargetDescription: ... } result
		// Meaning: Another element is preventing pointer events.
		//
		// It's safe to count an object return as an interception.
		// We just don't interpret what is intercepting with the target element
		// because we don't need any more functionality from this JS function
		// right now. 
		return false, errorFromDOMError("error:intercept")
```

Reply:
OK, clear.